### PR TITLE
Post.refactor dev

### LINKED ISSUE #38: Fix an issue with the image for create post form

### LINKED ISSUE #43: Refactor post service to follow a CRUD layout
While discussing with @BenjaminCDP about services, I noticed our post implementation wasn't the best

- We don't follow the crud layout
- The service should be rename post under services/post/post
- We store the postId in the post while it's not necessary
- The upload of the picture doesn't work
- The function used to upload picture should be in it's own service, under "services/storage/storage

@TheoRodriguess  and @FeuilleCiseaux , I'm currently working  on [the message branch](https://github.com/AgileFrontend/AgileFrontend/tree/dev-messgae) with benjamin help for the CRUD layout.

What i propose we do is open a new branch to follow the same principle I'm using on this branch, we'll see together this week how approach this

Once it's all working and the PR is merge, we can open a new branch to improve the current forms for post

### PR DESCRIPTION
**Pull request by Hendercraft**
-Add post service to mange all activities related to post in the back
-Refactor service storage to only manage firebase storage element
-Refactor create.post component to adjust to new services
-Bugfix : image now upload correctly, and their link are put in the post object in the db

### CODE REVIEW

**Comment by samaritan-mt** [comment]
Perfect was waiting for this to handle the share button

**Review by samaritan-mt — APPROVED** [review]

File: AgileFront/src/app/create.post/create.post.component.ts
Comment (BenjaminCDP) [inline AgileFront/src/app/create.post/create.post.component.ts:1]: Have you tried Event | undefined for event: any?

**Comment by Hendercraft** [comment]
The build was more thant 1Mb, the default max build size, modifed angular.json so it won't crash during bulding phase

**Review by Nellou90 — APPROVED** [review]

**Review by BenjaminCDP — APPROVED** [review]

File: AgileFront/src/app/create.post/create.post.component.ts
Comment (Hendercraft) [reply AgileFront/src/app/create.post/create.post.component.ts:1]: yep, since i access event.target, i get an error saying event does not implement it

**Comment by Hendercraft** [comment]
https://dev.to/bettercodingacademy/typescript-is-a-waste-of-time-change-my-mind-pi8
Literally me

**Comment by BenjaminCDP** [comment]
> https://dev.to/bettercodingacademy/typescript-is-a-waste-of-time-change-my-mind-pi8 Literally me

you're a whiny little bitch

**Review by BenjaminCDP — APPROVED** [review]

**Comment by Hendercraft** [comment]
Bro is about to get zuck by microsoft for saying bobo :
image

**Review by cyrpnt — APPROVED** [review]

**Pull request merged**